### ISSUE TITLE
If Docker is not running, the error message is vague/unhelpful: "Docker is running  containers, you need to switch to Windows containers"

### ISSUE THREAD

**Issue by dtkb** [microsoft/navcontainerhelper#1777]
**Describe the issue**
If Docker is not running, the error message should say so, instead of being vague/unhelpful. That is: if `$dockerOS` is empty, just say "Docker must be running" or something, instead of `Docker is running  containers, you need to switch to Windows containers.`.

**Scripts used to create container and cause the issue**
```
$containerName = 'somename'
$password = 'somepass'
$securePassword = ConvertTo-SecureString -String $password -AsPlainText -Force
$credential = New-Object pscredential 'admin', $securePassword
$auth = 'UserPassword'
$artifactUrl = Get-BcArtifactUrl -type 'Sandbox' -country 'gb' -select 'Latest'
New-BcContainer `
    -accept_eula `
    -containerName $containerName `
    -credential $credential `
    -auth $auth `
    -artifactUrl $artifactUrl `
    -imageName 'somename' `
    -updateHosts
```

**Full output of scripts**
```none
BcContainerHelper version 2.0.6
BcContainerHelper is version 2.0.6
BcContainerHelper is running as administrator
Host is Microsoft Windows 10 Pro - 2004
Docker is running  containers, you need to switch to Windows containers.
At C:\Program Files\WindowsPowerShell\Modules\BcContainerHelper\2.0.6\ContainerHandling\New-NavContainer.ps1:463 char:9
+         throw "Docker is running $dockerOS containers, you need to sw ...
+         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : OperationStopped: (Docker is runni...ows containers.:String) [], RuntimeException
    + FullyQualifiedErrorId : Docker is running  containers, you need to switch to Windows containers.
```

**Comment by freddydk** [comment]
That's strange.
Right above that line, the following code is:
```
    $dockerService = (Get-Service docker -ErrorAction Ignore)
    if (!($dockerService)) {
        throw "Docker Service not found. Docker is not started, not installed or not running Windows Containers."
    }

    if ($dockerService.Status -ne "Running") {
        throw "Docker Service is $($dockerService.Status) (Needs to be running)"
    }
```
Strange that the docker service is running but docker OS is empty, but I can see that dockerOS can be empty.
I will add this error is dockerOS is empty:

"Docker is not running or is wrongly configured"

**Comment by dtkb** [comment]
Thanks!

I don't think my situation is particularly strange, but then, Docker for Windows is always a strange experience throughout ;-)

This happens if I have booted the PC but not run Docker Desktop. Maybe the service is running but containers aren't? I dunno... If I can, I'll check its status on next boot for you.

Thanks for all your work on the ContainerHelpers!

**Comment by freddydk** [comment]
Might be, thanks

**Comment by freddydk** [comment]
Error message will be "Docker service is not yet ready." instead

**Comment by freddydk** [comment]
Shipped in 2.0.7
